### PR TITLE
selection plans fetching endpoint refactoring

### PR DESCRIPTION
**Pull request by romanetar**
ref https://tipit.avaza.com/project/view#!tab=task-pane&groupby=MyTaskDueDate&view=vertical&task=2860686&fileview=grid

Signed-off-by: romanetar <roman_ag@hotmail.com>

### CODE REVIEW

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSelectionPlansApiController.php [hunk 1]
```diff
@@ -327,6 +336,64 @@ public function getCurrentSelectionPlanByStatus($summit_id, $status)
...
+                        $filter->addFilterCondition(FilterElement::makeEqual('summit_id', $summit_id));
+                    }
+                    return $filter;
+                },
+                function () {
+                    return SerializerRegistry::SerializerType_Public;
+                },
+                null,
+                null,
+                function ($page, $per_page, $filter, $order, $applyExtraFilters) {
```
Comment (smarcet) [inline app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSelectionPlansApiController.php:375]: this is not necessary
the default implementation of the trait already calls this
https://github.com/OpenStackweb/summit-api/blob/f0515686ccf1d21fbaa0691fab78e5aecba89f75/app/Http/Controllers/Apis/Protected/Summit/Traits/ParametrizedGetAll.php#L45

**Review by smarcet — CHANGES_REQUESTED** [review]
@romanetar please review comments
regards

File: app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSelectionPlansApiController.php (hunk 1 above)
Comment (romanetar) [reply app/Http/Controllers/Apis/Protected/Summit/OAuth2SummitSelectionPlansApiController.php]: This was changed as requested

**Review by smarcet — APPROVED** [review]
looks good to me

**Pull request merged**